src/app/core/components/directives/details-height/details-height.directive.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  Directive, ElementRef, Inject, Input, OnDestroy, OnInit,
+} from '@angular/core';
+import { WINDOW } from 'app/helpers/window.helper';
+
+@Directive({
+  selector: '[ixDetailsHeight]',
+})
+export class IxDetailsHeightDirective implements OnInit, OnDestroy {
+  @Input() ixDetailsHeightParantClass: string;
+  @Input() hiddenHeight = 230;
+  @Input() maxShowHeight = 150;
```

Comment:
Not immediately obvious what these are. Can we add more description or rename?
Also I'm assuming one of these is responsible for getting height of the header. If so, can we query for that height (maybe in parent component)? Trying to make sure this doesn't break when header height changes.

Reply:
Agreed. Now these 2 parameters have better names, and adjusted default values

Reply:
Okay, but can we avoid magic numbers and query for height in the parent?